### PR TITLE
Merge `feature/active-pacemaker` to `master`

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR merges the Active Pacemaker feature branch to `master`, after which we can delete the branch. 
- In the unlikely case of a need to revert to pre-Jolteon consensus during the current spork, we have `v0.29-no-pacemaker`

### Files with merge conflicts
```
CONFLICT (content): Merge conflict in cmd/Dockerfile
CONFLICT (content): Merge conflict in consensus/hotstuff/eventhandler/event_handler_test.go
CONFLICT (content): Merge conflict in consensus/hotstuff/integration/instance_test.go
CONFLICT (content): Merge conflict in consensus/integration/integration_test.go
CONFLICT (content): Merge conflict in engine/common/follower/engine.go
CONFLICT (content): Merge conflict in engine/consensus/compliance/core.go
CONFLICT (content): Merge conflict in engine/execution/computation/computer/computer_test.go
CONFLICT (content): Merge conflict in go.mod
CONFLICT (content): Merge conflict in insecure/go.mod
CONFLICT (content): Merge conflict in integration/go.mod
CONFLICT (content): Merge conflict in module/metrics.go
CONFLICT (content): Merge conflict in module/metrics/noop.go
CONFLICT (content): Merge conflict in storage/mock/headers.go
CONFLICT (content): Merge conflict in storage/mocks/storage.go
```
### Additional Commits
- https://github.com/onflow/flow-go/pull/3895/commits/5277860ed25065aeb449a5d140c13b70bc2335c7 (`make tidy`)
- https://github.com/onflow/flow-go/pull/3895/commits/aa61441b20838b6f003b7b467d840bd291d9fee6 (`make generate-mocks`)
- https://github.com/onflow/flow-go/pull/3895/commits/7908418030005f098f964cf602cf692bdfbd424c https://github.com/onflow/flow-go/pull/3895/commits/0110499bf4b82a721bef5a7ba0990f63cc87f669 https://github.com/onflow/flow-go/pull/3895/commits/39bc9b0c815892ee4c781a6a53588ef0de09a68a (pin version of `flow-emulator` which includes PRs below)
  - https://github.com/onflow/flow-emulator/pull/275
  - https://github.com/onflow/flow-emulator/pull/306
  - https://github.com/onflow/flow-emulator/pull/311
- https://github.com/onflow/flow-go/pull/3895/commits/bb91d34cbc1caeeec807dbe96c15ccb15a7b7a30 (`make tidy` again)
- https://github.com/onflow/flow-go/pull/3895/commits/8e71b4350b25a054f5bad4dbd20dbd26c09cfdc2 Merge `master`, resolve new conflicts in `engine/execution/computation/computer/computer_test.go`
- https://github.com/onflow/flow-go/pull/3895/commits/d75481c2b41769c912858eb4bbbce33ae5d09131 Pin version of `onflow/flow` which includes Header model changes for Pacemaker
- https://github.com/onflow/flow-go/commit/f46be69140ab1af6c9501e13676f872572242ccd Fix header whitespace (`goimports`)

See [7e24c7db...bb91d34c](https://github.com/onflow/flow-go/pull/3895/files/7e24c7db89f89e4b250415fd35dbce0618213434..bb91d34cbc1caeeec807dbe96c15ccb15a7b7a30), [d75481c2...f46be691](https://github.com/onflow/flow-go/pull/3895/files/d75481c2b41769c912858eb4bbbce33ae5d09131..f46be69140ab1af6c9501e13676f872572242ccd) for a diff of all post-merge changes.

### CODE REVIEW

**Comment by pattyshack** [comment]
(the fvm change looks fine.  i'll remove myself as reviewer since i have no context)

File: consensus/hotstuff/eventhandler/event_handler_test.go [hunk 1]
```diff
@@ -13,263 +15,224 @@ import (
...
 // BlockProducer mock will always make a valid block
-type BlockProducer struct{}
-
-func (b *BlockProducer) MakeBlockProposal(qc *flow.QuorumCertificate, view uint64) (*model.Proposal, error) {
-	return createProposal(view, qc.View), nil
+type BlockProducer struct {
+	proposerID flow.Identifier
 }
 
+// //////// master
```
Comment (AlexHentschel) [inline consensus/hotstuff/eventhandler/event_handler_test.go:235]:
```suggestion
```

File: consensus/hotstuff/eventhandler/event_handler_test.go [hunk 2]
```diff
@@ -309,6 +272,19 @@ func (v *DenylistValidator) ValidateProposal(proposal *model.Proposal) error {
 	return nil
 }
 
+////////// master
+
```
Comment (AlexHentschel) [inline consensus/hotstuff/eventhandler/event_handler_test.go:276]:
```suggestion
```

**Comment by AlexHentschel** [comment]
could we please remove this white-space between the two 3rd-party imports?

**Review by AlexHentschel — APPROVED** [review]
Thanks for taking this on.

**Review by durkmurder — APPROVED** [review]

File: consensus/hotstuff/eventhandler/event_handler_test.go (hunk 2 above)
Comment (jordanschalm) [reply consensus/hotstuff/eventhandler/event_handler_test.go]: https://github.com/onflow/flow-go/pull/3895/commits/a421cfd5e31a4bdebddb366e99323ff25cb51b1c

File: consensus/hotstuff/eventhandler/event_handler_test.go (hunk 1 above)
Comment (jordanschalm) [reply consensus/hotstuff/eventhandler/event_handler_test.go]: https://github.com/onflow/flow-go/pull/3895/commits/a421cfd5e31a4bdebddb366e99323ff25cb51b1c

**Comment by jordanschalm** [comment]
> could we please remove this white-space between the two 3rd-party imports?

@AlexHentschel  This comment isn't associated with a line so I'm not sure where you're referring to here.

**Comment by jordanschalm** [comment]
Fixed import whitespace: https://github.com/onflow/flow-go/pull/3895/commits/f46be69140ab1af6c9501e13676f872572242ccd

**Comment by jordanschalm** [comment]
bors merge

**Comment by jordanschalm** [comment]
bors retry

**Pull request merged**